CoulombsLaw/coulombmain.py

```diff
@@ -0,0 +1,50 @@
+"""
+Electric Force Calculation using Coulomb's Law
+
+This module calculates the electric force between two point charges based on 
+user input, using Coulomb's law.
+"""
+
+import coulomb
+
+# Conversion factor: 1 C = 1,000,000 µC
+COULOMB_TO_MICROCOULOMB = 10**-6
+
+
+def main():
+    """
+    Main function to calculate and display the electric force between 
+    two point charges using Coulomb's law.
+    
+    This function:
+    1. Prompts the user to input the values of two charges (in microcoulombs) 
+    and the distance between them (in centimeters).
+    2. Converts the charges from microcoulombs to coulombs.
+    3. Converts the distance from centimeters to meters.
+    4. Calls the Coulomb's law function to compute the electric force between the charges.
+    5. Displays the converted charges (in coulombs) and the calculated force (in newtons).
+    """
+    #
+
+    # Get the first charge (q1) from the user, in coulombs.
+    q1 = float(input("Enter charge q1 (in microcoulombs): "))
+    # Get the second charge (q2) from the user, in coulombs.
+    q2 = float(input("Enter charge q2 (in microcoulombs): "))
+
+    # Use map and lambda to convert to microcoulombs
+    q1, q2 = map(lambda q: q * COULOMB_TO_MICROCOULOMB, [q1, q2])
+
+    # Get the distance (r) between the charges from the user, in meters.
+    r = float(input("Enter distance r (in centimeters): "))
+    # Calculate the electric force between the two charges using Coulomb's law.
+    result = coulomb.calculate_electric_force(q1, q2, r)
+    # Display the converted charges
+    print(f"q1 in microcoulombs: {q1} µC")
+    print(f"q2 in microcoulombs: {q2} µC")
+
+    # Display the calculated electric force in newtons (N).
+    print("The electric force between q1 and q2 is ", result, "N")
+
+
+if __name__ == "__main__":
```

Comment:
So the point of the __main__ loop is to give you a way to combine a python script and a module (see: https://docs.python.org/3/tutorial/modules.html#executing-modules-as-scripts). So having a separate “main” function is not necessary. In this case actually the whole coulombmain.py is not necessary, but you can add the content of the “main” function to a `__main__` function in coulomb.py to get the same functionality. However, since your coulombmain is mostly a function to get the input it probably would make sense to convert it to a command line parser (see: https://docs.python.org/3/library/argparse.html) since interactive input in python nowadays is not very popular and most likely done via a jupyter notebook.